pkg/datastore/deploymentstore.go

```diff
@@ -279,6 +282,30 @@ func (s *deploymentStore) UpdateStageMetadata(ctx context.Context, deploymentID,
 	})
 }
 
+func (s *deploymentStore) UpdateSharedMetadata(ctx context.Context, id string, metadata map[string]string) error {
+	return s.update(ctx, id, func(d *model.Deployment) error {
+		if d.MetadataV2.Shared == nil {
+			d.MetadataV2.Shared = &model.DeploymentMetadata_KeyValues{}
+		}
+		d.MetadataV2.Shared = &model.DeploymentMetadata_KeyValues{
+			KeyValues: mergeMetadata(d.MetadataV2.Shared.KeyValues, metadata),
+		}
+		return nil
+	})
+}
```

Comment:
It's better to use the generated getter method, which handles a nil pointer, so we don't have to initialize `d.MetadataV2.Shared`.

```suggestion
func (s *deploymentStore) UpdateSharedMetadata(ctx context.Context, id string, metadata map[string]string) error {
	return s.update(ctx, id, func(d *model.Deployment) error {
		d.MetadataV2.Shared = &model.DeploymentMetadata_KeyValues{
			KeyValues: mergeMetadata(d.GetMetadataV2().GetShared().GetKeyValues(), metadata),
		}
		return nil
	})
}
```

Reply:
thank you, I fixed : [befcc98](https://github.com/pipe-cd/pipecd/pull/6037/commits/befcc985239f8a4efe71d1647657dbe3bdf7090d)